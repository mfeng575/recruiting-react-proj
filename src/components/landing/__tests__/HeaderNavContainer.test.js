import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';
import {HeaderNavContainer} from '../HeaderNavContainer';


describe('HeaderNavContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = { apiCallsInProgress: 0 }
        const wrapper = shallow(<HeaderNavContainer {...props}/>);

        expect(wrapper).toHaveLength(1);

        expect(wrapper.find('nav')).toHaveLength(1);
        expect(wrapper.find('NavLink')).toHaveLength(2);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();          
    });
});
