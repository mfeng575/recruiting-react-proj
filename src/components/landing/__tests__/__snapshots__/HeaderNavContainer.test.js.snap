// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavContainer.test.js renders without crashing 1`] = `
<nav
  className="navbar navbar-toggleable-sm bg-info navbar-inverse"
>
  <div
    className="container"
  >
    <button
      className="navbar-toggler"
      data-target="#mainNav"
      data-toggle="collapse"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="mainNav"
    >
      <div
        className="navbar-nav"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/leaderboard"
        >
          PGA Leader Board
        </NavLink>
        <span
          className="ml-5"
        />
      </div>
    </div>
  </div>
</nav>
`;
